www/src/pages/blog/astro-019.astro

```diff
@@ -0,0 +1,171 @@
+
+          ```js
+          // src/pages/posts/[id].astro
+          // Tell Astro what pages to build for your route "/pages/:id"
+          export async function getStaticPaths() {
+            // Lets fetch posts from a Wordpress CMS:
+            const CSS_TRICKS_CMS = 'https://css-tricks.com/wp-json/wp/v2/posts?per_page=12&_embed';
+            const allPosts = await fetch(CSS_TRICKS_CMS).then(r => r.json());
+            // Then, create a new page from every post:
+            return allPosts.map((post) => ({
+              // Set the URL param "id" for the page URL
+              params: {id: post.id},
+              // Pass the post object as a prop to the page
+              props: {post},
+            }));
+          }
+          ```
+
+          To see more examples, check out this great [Wordpress Headless CMS project](https://github.com/chriscoyier/astro-css-trickzz) from Chris Coyier, or this [Shopify Ecommerce example](https://github.com/cassidoo/shopify-react-astro) by Cassidy Williams. 
+          
+          If you were a user of our original Collections API, this new file-based routing system completely replaces Collections with plenty of friendly warning messages to help you upgrade. We hope this new API solves some of the more common problems that users experienced with Collections. 
+
+          ## `client:only` loading directive
+
+          Sometimes, a UI component can't render on the server. Maybe it's because you've hit a bug in one of your dependencies, or maybe you're just using a library like D3 that can't run without the browser's `window` object. 
+          
+          You can usually wrap browser-only code with your own static server-side fallback UI. However several users wanted a way to render browser-only components without the boilerplate of additional placeholder HTML. Enter `client:only`.
+
+          Astro v0.19 ships with the new `client:only` loading directive to hydrate your component in the browser without server-side rendering. This provides a simple, straightforward fallback for any browser-only components.
+
+          ```html
+          <!-- only renders in the browser, no placeholder HTML -->
+          <MyReactComponent client:only />
+          ```
+
+          In a future release, we plan to add framework auto-detection (needed for multi-framework projects) and explore a more friendly API for generating fallback HTML for browser-only components.
+
+          Thank you Tony Sull (of [Navillus](https://navillus.dev/)) for contributing this new feature. 
+
+          ## `Astro.resolve()`
+
+          Astro 0.19 includes a new [`Astro.resolve()`](https://docs.astro.build/reference/api-reference#astroresolve) helper function to resolve relative file references in your templates. With this new function, you can reference images by relative path on disk and Astro will return the correct URL for the browser.
```

Comment:
nit: Above `v0.19` was used and now it's `0.19` without the v, should be consistent